### ISSUE TITLE
Unable to connect Rundeck with Vault (self signed certificates) 

### ISSUE THREAD

**Issue by eduardez** [rundeck/rundeck#8406]
**Describe the bug**
I´m unable to connect Rundeck with Vault via https using self signed certificates. Everytime I try to access I get this error:
`Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: 
sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification 
path to requested target`

**My Rundeck detail**

* Rundeck version: 4.12
* install type: docker
* OS Name/version: RHEL
* DB Type/version: postgres

**To Reproduce**
1. Configure Vault HTTPS with a self signed certificate.
2. Configure Rundeck to use Vault via HTTPS
3. Try to access key storage
4. Error

**Expected behavior**
Successful connection with Vault.

**Comment by MegaDrive68k** [comment]
Hi @eduardez,

Did you try adding the vault certificate to the [Java cacert](https://stackoverflow.com/a/11617655/10426011) (Rundeck)?

Regards.

**Comment by eduardez** [comment]
Hi @MegaDrive68k 

Yes, just before running rundeck I add them to the truststore. Here is my ssl.properties.
``` 
keystore=/home/rundeck/etc/keystore
keystore.password=<supersecretpass>
key.password=<supersecretpass>
truststore=/home/rundeck/etc/truststore
truststore.password=<supersecretpass>
```
And just before starting rundeck, I run this script:
``` bash
echo "-----BEGIN CERTIFICATE-----
MIIEczCCA1...
...trDYN+rI
-----END CERTIFICATE-----
" > /tmp/auto.crt

echo "supersecretpass" | sudo -S keytool -import -trustcacerts -noprompt -alias automation -keystore /home/rundeck/etc/keystore -file /tmp/auto.crt -keypass supersecretpass -storepass supersecretpass
echo "supersecretpass" | sudo -S keytool -import -trustcacerts -noprompt -alias automation -keystore /home/rundeck/etc/truststore -file /tmp/auto.crt -keypass supersecretpass -storepass supersecretpass
```

**Comment by fdevans** [comment]
Thanks for the report. As part of a cleanup of our issue tracker, we're closing issues older than 12 months, in line with Rundeck's support policy of one year from a release's date (see the [release calendar](https://docs.rundeck.com/docs/history/release-calendar.html) for currently supported versions). This isn't a judgment that the report is invalid — many of these are likely still real — but we can't verify against a version this old.

If you're still seeing this on a currently supported release (6.1.x or 5.20.x+), please open a new issue using our [bug report template](https://github.com/rundeck/rundeck/issues/new?template=bug_report.yml) with your version and reproduction steps. Please don't reopen this issue — a fresh issue keeps things tracked against a supported version and gets it in front of us properly. We're actively working on improving how issues get submitted and triaged going forward.

Thanks for your patience,
The Rundeck Team
